--- a/lib/modules.nix
+++ b/lib/modules.nix
@@ -149,7 +149,6 @@

         config = {
           _module.args = {
-            inherit extendModules;
             moduleType = type;
           } // args;
         };
@@ -159,7 +158,7 @@
         let collected = collectModules
           (specialArgs.modulesPath or "")
           (modules ++ [ internalModule ])
-          ({ inherit lib options config specialArgs; } // specialArgs);
+          ({ inherit lib options config extendModules specialArgs; } // specialArgs);
         in mergeModules prefix (reverseList collected);

       options = merged.matchedOptions;
